test/widgets/lightbox_test.dart

```diff
@@ -152,5 +267,148 @@ void main() {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      final halfTime = FakeVideoPlayerPlatform.kTestVideoDuration * 0.5;
+
+      await tester.pump(halfTime);
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, halfTime);
+      check(FakeVideoPlayerPlatform.position).equals(halfTime);
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+
+      // Near the end of the video.
+      await tester.pump(halfTime - const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester,
+        FakeVideoPlayerPlatform.kTestVideoDuration - const Duration(milliseconds: 500));
+      check(FakeVideoPlayerPlatform.position)
+        .equals(FakeVideoPlayerPlatform.kTestVideoDuration - const Duration(milliseconds: 500));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+
+      // At exactly the end of the video.
+      await tester.pump(const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.position).equals(FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.isPlaying).isFalse(); // stopped playing
+
+      // After the video ended.
+      await tester.pump(const Duration(milliseconds: 500));
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.position).equals(FakeVideoPlayerPlatform.kTestVideoDuration);
+      check(FakeVideoPlayerPlatform.isPlaying).isFalse();
+    });
+
+    testWidgets('ensure \'seekTo\' is called only once', (tester) async {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      const padding = 24.0;
+      final rect = tester.getRect(find.byType(Slider));
+      final trackWidth = rect.width - padding - padding;
+      final trackStartPos = rect.centerLeft + const Offset(padding, 0);
+      final twentyPercent = trackWidth * 0.2; // 20% increments
+
+      // Verify the actually displayed current position at each
+      // gesture increments.
+      final gesture = await tester.startGesture(trackStartPos);
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, Duration.zero);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.2);
```

Comment:
nit:
```suggestion
      checkPosition(tester, 0.2 * FakeVideoPlayerPlatform.kTestVideoDuration);
```

That's the standard convention for mathematicians and physicists — one would write $0.2 \cdot T$ or $0.2 \\, T$, and rarely to never $T \cdot 0.2$ or $T \\, 0.2$.

Here, it's helpful also just by foregrounding the part that's more interesting, so it isn't inconspicuously off at the end of a longish line.

Reply:
Unfortunately, I was unable to get this working with operator overloading extensions, local operator overloads on `num`/`double` won't take precedence over the upstream ones.
https://github.com/dart-lang/language/issues/966

Reply:
Oh I see, I forgot that this is a Duration and not a number. So be it, then.

Reply:
In any case, for now I've reduced the character count:
```diff
-       checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.4);
+       checkPosition(tester, kTestVideoDuration * 0.4);
```